plat/meson/gxbb/gxbb_pm.c

```diff
@@ -113,17 +130,54 @@ static void gxbb_pwr_domain_off(const psci_power_state_t *target_state)
 
 	gicv2_cpuif_disable();
 
+	/* CPU0 can't be turned OFF/ON, emulate it with a WFE loop */
+	if (core == GXBB_PRIMARY_CPU)
+		return;
+
 	scpi_set_css_power_state(mpidr,
 				 SCPI_POWER_OFF, SCPI_POWER_ON, SCPI_POWER_ON);
 }
 
+static void __dead2 gxbb_pwr_domain_pwr_down_wfi(const psci_power_state_t
+						 *target_state)
+{
+	unsigned int core = plat_gxbb_calc_core_pos(read_mpidr_el1());
+
+	/* CPU0 can't be turned OFF/ON, emulate it with a WFE loop */
+	if (core == GXBB_PRIMARY_CPU) {
+		VERBOSE("BL31: CPU0 entering wait loop...\n");
+
+		while (gxbb_cpu0_go == 0)
+			wfe();
+
+		VERBOSE("BL31: CPU0 resumed.\n");
+
+		gxbb_cpu0_go = 0;
+		dsb();
+		isb();
+
+		disable_mmu_el3();
```

Comment:
Issuing an rmr_el3 to reset the core might work better. Also the reset of the variable 
`gxbb_cpu0_go` to 0 can be done in the `on_finish` hook since the CPU_ON call can still get through and set the variable to 1. Since the `on_finish` handler executes after the PSCI affinity info state of the CPU has been updated to RUN, the CPU_ON call can no longer reach the platform layer. This will prevent spurious wake-ups.

Reply:
You're right, I didn't even remember that register.